src/monattack.cpp

```diff
@@ -2255,6 +2255,172 @@ void mattack::tentacle(monster *z, int index)
+                                                      _("The %s's arms fly out at <npcname>, but they dodge!"),
+                                            z->name().c_str() );
+            }
+            if( !uncanny ) {
+                target->on_dodge( z, z->type->melee_skill * 2 );
+            }
+        } else if( seen ) {
+            add_msg( _("The %s's arms fly out at %s, but they miss and sail past!"), z->name().c_str(), target->disp_name().c_str() );
+        }
+        return;
+    }
+
+    const int dir = g->m.coord_to_angle( z->posx(), z->posy(), target->posx(), target->posy() ); //Orientation of where you'll end up
+    tileray tdir(dir);
+    int range = (z->type->melee_sides * z->type->melee_dice) / 10;
+    tripoint pt = target->pos();
+    while (range > 0) {
+        if( foe != nullptr ) {
+        tdir.advance();
+        pt.x = z->posx() + tdir.dx();
+        pt.y = z->posy() + tdir.dy();
+        if( target->is_player() && ( pt.x < SEEX * int(MAPSIZE / 2) || pt.y < SEEY * int(MAPSIZE / 2) ||
+            pt.x >= SEEX * (1 + int(MAPSIZE / 2)) || pt.y >= SEEY * (1 + int(MAPSIZE / 2)) ) ) {
+            g->update_map( pt.x, pt.y );
+        }
+        if (foe->in_vehicle) {
+            g->m.unboard_vehicle(foe->pos());
+        }
+        foe->setpos( pt );
+    }
+    else {
+        zz->setpos( pt );
+    }
+        range--;
+        if( seen ) {
+            g->draw();
+        }
+    }
+    if( seen ) {
+        add_msg( _("The %s's arms fly out and pull %s back!"), z->name().c_str(), target->disp_name().c_str() );
+    }
+}
+
+void mattack::grab(monster *z, int index)
+{
+    if( !z->can_act() ) {
+        return;
+    }
+    Creature *target = z->attack_target();
+    if( target == nullptr || rl_dist( z->pos(), target->pos() ) > 1 ) {
+        return;
+    }
+
+    player *foe = dynamic_cast< player* >( target );
+    bool seen = g->u.sees( *z );
+
+    z->reset_special(index); // Reset timer
+    z->moves -= 80;
+    bool uncanny = target->uncanny_dodge();
+    if( uncanny || dodge_check(z, target) ){
+        if( foe != nullptr ) {
+            if( seen ) {
+                auto msg_type = foe == &g->u ? m_warning : m_info;
+                target->add_msg_player_or_npc( msg_type, _("The %s gropes at you, but you dodge!"),
+                                                      _("The %s gropes at <npcname>, but they dodge!"),
+                                            z->name().c_str() );
+            }
+            if( !uncanny ) {
+                target->on_dodge( z, z->type->melee_skill * 2 );
+            }
+    } else if( seen ) {
+        add_msg( _("The %s gropes at %s, but misses!"), z->name().c_str(), target->disp_name().c_str() );
+    }
+    return;
+    }
+    target->add_msg_player_or_npc(m_bad, _("%s grabs you!"), _("%s grabs <npcname>!"),
+                                z->disp_name().c_str());
+    if (foe == nullptr){
```

Comment:
Yeah, that would fit well too. You could even unify them into one function: grab resistance. It could return some number that could be used both for grabbing and dragging, because in most cases dragging involves grabbing in some way.

But if you want to keep them separate, it would be good if the drag check wasn't just a drag check, but something like stability check that would protect not just from dragging, but also knockback. They're quite similar, just that knockback generally involves more force.